app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,54 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.order(market_name: :asc)
+    @all_markets = Market.order(market_name: :asc)
+  end
+
+  def show
+    @market = Market.find(params[:id])
```

Comment:
We just learned about before actions today, as we continue work on bEtsy. Thank you for the suggestion re: refactoring. I have a hard time seeing where that should happen, so I really appreciate the note.